File: java/figuras/src/main/java/com/maicol/figuras/figureGeometric.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.maicol.figuras;

/**
 *
 * @author Maicol
 */  
public abstract class figureGeometric {
    private String name;
    private String spanishName;

    public figureGeometric(String name, String spanishName) {
        this.name = name;
        this.spanishName = spanishName;
    }

    public String getName() {
        return name;
    }

    public String getSpanishName() {
        return spanishName;
    }

    public abstract double calculateArea();
}

File: php/examples/figuras/php/figureGeometric.php
<?php


abstract class FigureGeometric{
    private string $name;
    private string $spanishName;
    function __construct(string $name, string $spanishName){
        $this->name = $name;
        $this->spanishName = $spanishName;
    }
    public function getName():string{
        return $this->name;
    }
    public function getSpanishName():string{
        return $this->spanishName;
    }
    abstract public function calculateArea():float; 
}
?>